### ISSUE TITLE
CATTLE_FORWARD chain gets reordered on docker restart

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#9503]
**Rancher versions:**
rancher/server: 1.3.1

Problem:
On docker restart, CATTLE_FORWARD chain is not the first rule in FORWARD chain, ideally rancher agent can detect broken iptables rules and fix them.

**Comment by niusmallnan** [comment]
I think this might be useful. `Fix bug for insert base iptables rules`https://github.com/rancher/plugin-manager/pull/72/commits

**Comment by deniseschannon** [comment]
Duplicate of https://github.com/rancher/rancher/issues/8978

**Comment by leodotcloud** [comment]
8978 is related to CATTLE_NETWORK_POLICY chain, not CATTLE_FORWARD. They are similar.

**Comment by leodotcloud** [comment]
Fixed in `network-services:v0.2.7` by `network-manager:v0.7.10`

**Comment by sangeethah** [comment]
Rancher Server version - Build from v1.6
rancher/network-manager:v0.7.10 and rancher/net:v0.13.0

After restart of docker , CATTLE_NETWORK_POLICY and CATTLE_FORWARD forward chain get programmed in the top after sometime.

As soon as docker restarts , CATTLE_NETWORK_POLICY and CATTLE_FORWARD forward chain get programmed in the bottom of the chain but after about a maximum of 2 minutes , this gets autocorrected and CATTLE_NETWORK_POLICY and CATTLE_FORWARD forward chain gets programmed at the top of the forward chain.
```
root@santest161:~# iptables -L FORWARD -n -v
Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           
25122   10M CATTLE_NETWORK_POLICY  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
25122   10M CATTLE_FORWARD  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
root@santest161:~# iptables -L FORWARD -n -v
Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
 3732 2132K CATTLE_NETWORK_POLICY  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 3732 2132K CATTLE_FORWARD  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 1398  305K DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
  428  185K DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
  421  184K ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
  970  120K ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    7   420 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           
root@santest161:~# 
```
